src/main/java/net/smartcosmos/usermanagement/role/service/UpdateRoleServiceDefault.java

```diff
@@ -66,25 +69,25 @@ public void update(
         }
     }
 
-    private ResponseEntity<?> updateRoleWorker(String roleUrn, RoleRequest restRequest, SmartCosmosUser user) {
+    private ResponseEntity<?> updateRoleWorker(String roleUrn, RoleRequest updateRoleRequest, SmartCosmosUser user) {
 
-        final RoleRequest updateRoleRequest = conversionService
-            .convert(restRequest, RoleRequest.class);
-
-        Optional<RoleResponse> updateRoleResponse = roleDao.updateRole(user.getAccountUrn(), roleUrn, updateRoleRequest);
+        Optional<RoleResponse> updateRoleResponse;
+        try {
+            updateRoleResponse = roleDao.updateRole(user.getAccountUrn(), roleUrn, updateRoleRequest);
+        } catch (IllegalArgumentException e) {
```

Comment:
No. We had only the `Optional` before and couldn't differentiate between _Role not found_ and _Role name conflict_.